service/controller/v1/resource/node/create.go

```diff
@@ -31,19 +32,27 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
+	if customObject.Status.HasFinalCondition() {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "node config status already has final state")
+		resourcecanceledcontext.SetCanceled(ctx)
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource for custom object")
+
+		return nil
+	}
```

Comment:
I like it being on the top. It's like early return.

Reply:
It is. Only the type magic is above, which we need to have the type and to the method call.

Reply:
Yes it is. I realized. I just expressed my appreciation :)

Reply:
I see. Thanks buddy. <3